Library/Homebrew/formula_auditor.rb

```diff
@@ -289,6 +289,12 @@ def audit_deps
 
           next unless @core_tap
 
+          if dep_f.deprecated? && !formula.deprecated?
+            problem <<~EOS
+              Dependency '#{dep.name}' is deprecated. Formulae with dependents should not be deprecated.
```

Comment:
I think it would be good to clarify that the formula with the audit problem isn't the one that needs changing (possibly). Maybe something like this:

```suggestion
              Dependency '#{dep.name}' should be un-deprecated because it has dependents that are not deprecated.
```

Reply:
Or, I guess:

```suggestion
              Dependency '#{dep.name}' is deprecated but has un-deprecated dependents. Either un-deprecate '#{dep.name}' or deprecate its dependents including '#{formula.name}.
```

Reply:
Added a variation of your second suggestion.